source/slang/slang-emit-c-like.cpp

```diff
@@ -159,6 +159,14 @@ void CLikeSourceEmitter::emitDeclarator(DeclaratorInfo* declarator)
         }
         break;
 
+    case DeclaratorInfo::Flavor::Ref:
+        {
+            auto refDeclarator = (RefDeclaratorInfo*)declarator;
+            m_writer->emit("&");
+            emitDeclarator(refDeclarator->next);
+        }
+        break;
+
```

Comment:
I would have assumed we'd need additional cases here for pointer and function types...

Reply:
These are placed under the CPPEmit override. We can't get rid of that for now because of the existing logic of emitting `FixedArray<T>` and `Array<T>` in C++, and this is different from the default implementation.

Once we clean this up we should be able to just put all these logic in CLikeEmit. But before then I would just implement it in CppEmit to reduce the chance of running into issues.